lib/ramen.ex

```diff
@@ -1,18 +1,100 @@
+defmodule PullRequest do
+  defstruct [:title, :number, :participants]
+end
+
+defmodule Participant do
+  defstruct [:username]
+end
+
+defmodule Decoder do
+  def decode(payload, into: PullRequest) do
+    %{"title" => title, "number" => number} = payload
+
+    %PullRequest{title: title, number: number}
+  end
+
+  def decode(payload, into: Participant) do
+    payload
+    |> get_in(["data", "repository", "pullRequest", "participants", "edges"])
+    |> Enum.map(&get_in(&1, ["node", "login"]))
+    |> Enum.map(&%Participant{username: &1})
+  end
+end
+
+defmodule Ramen.Config do
+  defstruct [:token, :client, :get, :http_client]
+end
+
 defmodule Ramen do
-  @moduledoc """
-  Documentation for Ramen.
-  """
+  @spec new(String.t(), fun(), fun(), fun()) :: %Ramen.Config{}
+  def new(token, client, get, http_client) do
+    %Ramen.Config{
+      token: token,
+      client: client.(%{access_token: token}),
+      get: get,
+      http_client: http_client
+    }
+  end
 
-  @doc """
-  Hello world.
+  @spec list_pull_requests(String.t(), String.t(), %Ramen.Config{}) ::
+          {:ok, [%PullRequest{}]} | {:error, String.t()}
+  def list_pull_requests(owner, repository, config, opts \\ []) do
```

Comment:
vc criou responsabilidade dupla, puxar participants junto com pull request, como voce justifica isso?

Reply:
Realmente, meu racional foi não expor os métodos de `*_participants` e forçar o uso dele pelo `list_pull_requests` - mas não faz mto sentido